dali/pipeline/executor/executor.h

```diff
@@ -84,33 +88,59 @@ class DLL_PUBLIC Executor {
   DISABLE_COPY_MOVE_ASSIGN(Executor);
 
  protected:
-  using WorkspaceBlob = struct {
-    vector<HostWorkspace> cpu_op_data;
-    vector<MixedWorkspace> mixed_op_data;
-    vector<DeviceWorkspace> gpu_op_data;
-    vector<SupportWorkspace> support_op_data;
+  struct WorkspaceBlob {
+    workspace_owner_t op_data;
+
+    void Resize(int support, int cpu, int mixed, int gpu) {
+      std::get<static_cast<int>(DALIOpType::SUPPORT)>(op_data).resize(support);
```

Comment:
This hurts my eyes, really. If it were not an `enum class` the same line would look like this:
`std::get<DALI_OP_SUPPORT>(op_data).resize(support)`